libethcore/ChainOperationParams.h

```diff
@@ -62,11 +68,24 @@ struct ChainOperationParams
     /// The chain sealer name: e.g. Ethash, NoProof, BasicAuthority
     std::string sealEngineName = "NoProof";
 
+    // Example of how to check EIP activation from outside of EVM:
```

Comment:
Can we simplify this: 

    // Example of how to check EIP activation from outside of EVM:
    // bool isEIP2046Enabled(u256 const& _blockNumber) const
    // {
    //     return _blockNumber >= lastForkBlock && lastForkAdditionalEIPs.eip2046;
    // }
    // After hard fork finalization this is changed to:
    // bool isEIP2046Enabled(u256 const& _blockNumber) const
    // {
    //     return _blockNumber >= berlinForkBlock;
    // }

Instead of having a separate "is enabled" function for each EIP could we instead have a "isEipEnabled" function which takes both a block number and eip number and does the check? You'd need to add a switch statement in the function but overall I think it would result in cleaner code since you'd have fewer function signatures

Reply:
I wanted these functions to be very efficient (constant time), because they might be called in any hot path of consensus-critical code (only outside of EVM though). So minimizing the overhead of these helpers might be important for benchmarking EIP implementations and in general.

Implementing this with a switch would mean O(n) linear search for each such check, where n would grow with each new implemented EIP.